Adding Turkish language support

g-hano opened the pull request:
I created `turkish.py` and `turkish_bert.py` to support Turkish language. Used [dbmdz/bert-base-turkish-cased](https://huggingface.co/dbmdz/bert-base-turkish-cased) as tokenizer

g-hano committed:
Create turkish.py

g-hano committed:
Create turkish_bert.py

g-hano committed:
Update cleaner.py

g-hano committed:
Update __init__.py

g-hano committed:
Update symbols.py

g-hano committed:
Update utils.py

kadirnar commented:
@g-hano Did you train with Turkish data?

g-hano commented:
> @g-hano Did you train with Turkish data?

Yes I did, had to update many lines but finally was able to train on 4 T4s. Extra training on 46k audio samples made the model produce clearer outputs.

tosunozgun commented:
Hello, thank you for supporting the Turkish TTS model, first of all. When I reviewed your code, I noticed that you convert text to lowercase during normalization. Unfortunately, for Turkish, uppercase "I" is converted to lowercase "i," which is incorrect. As a solution, the following change is needed: text.replace("I", "ı").lower(). Since you are using lowercase text, I recommend trying ytu-ce-cosmos/turkish-base-bert-uncased.

g-hano committed:
Update turkish.py

Handle Turkish character 'I' and change tokenizer to ytu-ce-cosmos/turkish-base-bert-uncased

g-hano commented:
> Hello, thank you for supporting the Turkish TTS model, first of all. When I reviewed your code, I noticed that you convert text to lowercase during normalization. Unfortunately, for Turkish, uppercase "I" is converted to lowercase "i," which is incorrect. As a solution, the following change is needed: text.replace("I", "ı").lower(). Since you are using lowercase text, I recommend trying ytu-ce-cosmos/turkish-base-bert-uncased.

Thank you for your recommendations, I updated the code.

g-hano committed:
Update turkish_bert.py

g-hano committed:
fix model_id

furkeen commented:
Thanks for the great contribution @g-hano . I'm trying to test it however I couldn't find a way to test. Is there any way to test something like that? 
`melo "Merhaba ben Türkçe konuşabiliyorum." output.wav --language TR`

kadirnar commented:
I am developing the [MeloPlus](https://github.com/kadirnar/MeloPlus) library to add more features to the MeloTts library. I added Thai and Turkish language support.

@furkeen You should find a Turkish model trained with the melotts model.

faraday commented:
@kadirnar Does this need `Vyvo/MeloTTS-Ljspeech` to run properly?
That's not visible on Hugging Face.

karayakar commented:
Any huggingface space for testing? or sample output?
